src/extension.ts

```diff
@@ -47,8 +48,14 @@ export const loadEnvironments = async (
     extensionState.onFailedToGetBranch();
     return;
   }
-  await environmentsDataProvider.refresh();
-  extensionState.setIsLoading(false);
+  try {
+    await environmentsDataProvider.refresh();
+    startEnvironmentPolling();
+  } catch (e) {
+    onPullingEnvironmentError(e);
```

Comment:
I noticed that you use both "pull" and "poll". is it intentionally?

Reply:
291cc0901f841e0fed8d3c561e33cd2444491202